azure-communication-ui/azure-communication-ui/src/calling/java/com/azure/android/communication/ui/calling/redux/middleware/AudioFocusMiddleware.kt

```diff
@@ -0,0 +1,124 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.android.communication.ui.calling.redux.middleware
+
+import android.content.Context
+import android.media.AudioFocusRequest
+import android.media.AudioManager
+import android.os.Build
+import androidx.annotation.RequiresApi
+import com.azure.android.communication.ui.calling.redux.Dispatch
+import com.azure.android.communication.ui.calling.redux.Middleware
+import com.azure.android.communication.ui.calling.redux.Store
+import com.azure.android.communication.ui.calling.redux.action.Action
+import com.azure.android.communication.ui.calling.redux.action.CallingAction
+import com.azure.android.communication.ui.calling.redux.state.ReduxState
+
+// / Audio Focus Middleware
+// / Handles AudioFocus by requesting/releasing before the appropriate actions
+// / Rejects actions such as Start/Resume when Focus can't be retrieved
+internal interface AudioFocusMiddleware
+
+// Handle Audio Focus for different platforms
+// Provides ability to listen to the changes
+internal abstract class AudioFocusHandler : AudioManager.OnAudioFocusChangeListener {
```

Comment:
Adam, it will be great if we can use manager instead of middleware for android speific logic. As per previous team agreements all the Android specifc logic is handled in managers.

Reply:
seen as this has been already done, I'm going to resolve conversation and close for now.